language-support/ts/daml-ledger/index.ts

```diff
@@ -517,12 +536,30 @@ class Ledger {
       } else {
         console.error('Ledger.streamQuery unknown message', json);
       }
-    });
+    };
+    const onClose = ({ code, reason }: {
+        wasClean: boolean; code: number;
+        reason: string; target: WebSocket;
+    }): void => {
+      emitter.emit('close', {code, reason});
+      const now = new Date().getTime();
+      // we try to reconnect if we could connect previously and we were live for at least
+      // 'reconnectThreshold'.
+      if (lastOffset && isLiveSince && now - isLiveSince >= this.reconnectThreshold) {
+        console.log('reconnecting WebSocket');
+        isLiveSince = undefined;
+        isReconnecting = true;
```

Comment:
Let's better be safe than sorry:
```suggestion
        isReconnecting = true;
        ws.removeEventListener('open', onOpen);
        ws.removeEventListener('message', onMessage);
        ws.removeEventListener('close', onClose);
```